File: TTdsa-java/Exercise/src/com/aacirq/string/N43.java
package com.aacirq.string;

public class N43 {
    public static void main(String[] args) {
        N43 sol = new N43();

        // "XYZdefabc"
        System.out.println(sol.LeftRotateString("abcXYZdef", 3));
        // "a"
        System.out.println(sol.LeftRotateString("a", 3));
    }

    public String LeftRotateString(String str,int n) {
        if (str.length() == 0)
            return str;
        n %= str.length();
        return str.substring(n) + str.substring(0, n);
    }
}

File: TTdsa-cpp/Exercise/string/N43.cpp
//
// Created by renxin on 2021/6/10.
//

#include "ttdsa.h"

class Solution {
public:
    string LeftRotateString(string str, int n) {
        if (str.size() == 0)
            return str;
        n %= str.size();
        return str.substr(n, str.size() - n) + str.substr(0, n);
    }
};

int main() {
    Solution sol;

    // XYZdefabc
    cout << sol.LeftRotateString("abcXYZdef", 3) << endl;
    // a
    cout << sol.LeftRotateString("a", 3) << endl;
    // <empty line>
    cout << sol.LeftRotateString("", 3) << endl;

    return 0;
}